theano/sandbox/cuda/__init__.py

```diff
@@ -150,23 +132,92 @@ def try_import():
     # use already compiled GPU op and not the others.
     # Also, if cxx is not available, we need to disable all GPU code.
     set_cuda_disabled()
-    compile_cuda_ndarray = False
 elif not config.device.startswith('gpu') and config.force_device:
     # We where asked to NEVER use the GPU
     set_cuda_disabled()
-    compile_cuda_ndarray = False
-else:
-    # Add the theano cache directory's cuda_ndarray subdirectory to the
-    # list of places that are hard-coded into compiled modules' runtime
-    # library search list.  This works in conjunction with
-    # nvcc_compiler.NVCC_compiler.compile_str which adds this folder during
-    # compilation with -L and also adds -lcuda_ndarray when compiling
-    # modules.
-    nvcc_compiler.add_standard_rpath(cuda_ndarray_loc)
-    compile_cuda_ndarray = not try_import()
 
+# If $TMPDIR is defined, nvopencc wants it to exist
+if cuda_available and 'TMPDIR' in os.environ :
+    tmpdir = os.environ[ 'TMPDIR' ]
+    if not os.path.exists( tmpdir ) :
+        os.makedirs( tmpdir )
```

Comment:
This chunk of code had already been in **__ init __.py** prior to my pull request. I kept it to ensure compatibility.